File: frontend/src/types/listing.ts
import type { MarketplaceSource } from "./opportunity";

export interface ListingDetail {
  id: string;
  source: MarketplaceSource;
  externalId: string;
  url: string;
  title: string;
  description?: string;
  price: number;
  shippingCost: number;
  condition: string;
  imageUrls: string[];
  listedAt?: string;
  createdAt: string;
  projectedRestorerNet: number;
  projectedHarvestYield: number;
}

File: backend/app/schemas/listing.py
from datetime import datetime

from app.schemas.base import CamelModel
from app.schemas.common import MarketplaceSource


class ListingDetailResponse(CamelModel):
    id: str
    source: MarketplaceSource
    external_id: str
    url: str
    title: str
    description: str | None = None
    price: float
    shipping_cost: float
    condition: str
    image_urls: list[str]
    listed_at: datetime | None = None
    created_at: datetime
    projected_restorer_net: float
    projected_harvest_yield: float
